File: clients/ruby/model/server_message_game.rb
require_relative 'player_view'
class ServerMessageGame
    attr_accessor :player_view
    def initialize(player_view)
        @player_view = player_view
    end
    def self.read_from(stream)
        if stream.read_bool()
            player_view = PlayerView.read_from(stream)
        else
            player_view = nil
        end
        ServerMessageGame.new(player_view)
    end
    def write_to(stream)
        if @player_view.nil?
            stream.write_bool(false)
        else
            stream.write_bool(true)
            @player_view.write_to(stream)
        end
    end
end

File: clients/cpp/model/ServerMessageGame.cpp
#include "ServerMessageGame.hpp"

ServerMessageGame::ServerMessageGame() { }
ServerMessageGame::ServerMessageGame(std::shared_ptr<PlayerView> playerView) : playerView(playerView) { }
ServerMessageGame ServerMessageGame::readFrom(InputStream& stream) {
    ServerMessageGame result;
    if (stream.readBool()) {
        result.playerView = std::shared_ptr<PlayerView>(new PlayerView());
        *result.playerView = PlayerView::readFrom(stream);
    } else {
        result.playerView = std::shared_ptr<PlayerView>();
    }
    return result;
}
void ServerMessageGame::writeTo(OutputStream& stream) const {
    if (playerView) {
        stream.write(false);
    } else {
        stream.write(true);
        (*playerView).writeTo(stream);
    }
}
std::string ServerMessageGame::toString() const {
    return std::string("ServerMessageGame") + "(" +
        "TODO" + 
        ")";
}
